src/backend/marsha/core/serializers/video.py

```diff
@@ -266,13 +267,28 @@ def get_vod_urls(self, obj):
                 urls["thumbnails"][resolution] = thumbnail_urls.get(
                     resolution, video_storage.url(f"{base}/thumbnail.jpg")
                 )
-                urls["previews"] = video_storage.url(f"{base}/thumbnail.jpg")
-                urls["manifests"] = {
-                    "hls": video_storage.url(f"{base}/master.m3u8"),
-                }
+
+            urls["manifests"] = {
+                "hls": self._get_hls_url(base, obj),
+            }
+            urls["previews"] = video_storage.url(f"{base}/thumbnail.jpg")
 
         return urls
 
+    def _get_hls_url(self, base_url, video):
+        """
+        Compute the HLS url for a video. if the url method of the video storage has parameters
+        we use it to have a different cache depending the resolutions available.
+        """
+        storage_url_signature = signature(video_storage.url)
+        if storage_url_signature.parameters.get("parameters"):
+            return video_storage.url(
+                f"{base_url}/master.m3u8",
+                parameters={"v": hash(str(video.resolutions))},
+            )
+
+        return video_storage.url(f"{base_url}/master.m3u8")
```

Comment:
A test should be added nope?

Reply:
I added a test and I change how the hash is computed. The `hash` function is initialized with a seed, fo for the same resolutions the hash can change from a pod to an other one.